src/main/capi/duckdb_value-c.cpp

```diff
@@ -1,9 +1,22 @@
 #include "duckdb/main/capi/capi_internal.hpp"
+#include "duckdb/common/types/value.hpp"
+#include "duckdb/common/types.hpp"
 
+static duckdb_value WrapValue(duckdb::Value *list_value) {
```

Comment:
If this takes a `unique_ptr<duckdb::Value> &&value` we can better safeguard that `duckdb_value` doesn't contain stack memory.